src/main/scala/org/broadinstitute/dsde/firecloud/service/LibraryService.scala

```diff
@@ -95,4 +97,20 @@ class LibraryService (protected val userInfo: UserInfo, val rawlsDAO: RawlsDAO)(
       }
     }
   }
+
+  def setWorkspaceIsPublished(ns: String, name: String, value: Boolean): Future[PerRequestMessage] = {
+    rawlsDAO.getWorkspace(ns, name) flatMap { workspaceResponse =>
+      // verify owner on workspace
+      if (!workspaceResponse.accessLevel.contains("OWNER")) {
+        Future(RequestCompleteWithErrorReport(Forbidden, "must be an owner"))
+      } else {
+        val operations: Seq[AttributeUpdateOperation] = Seq()
+        // either add or remove the library attribute
+        if (value) operations :+ AddUpdateAttribute("library", AttributeString("published"))
+        else operations :+ RemoveAttribute("library")
```

Comment:
Creating one seq instead of doing it twice makes sense, thanks!

I think true/false sounds better to use for this case (especially if it's just a POST/DELETE with no additional parameters). 
Should there be a ticket for supporting boolean attributes?